File: modules/protocol/769/src/main/java/candle/protocol/codec/MinecraftDataCodec769.java
package candle.protocol.codec;

import candle.types.resource.Identifier;

public class MinecraftDataCodec769 extends AbstractDataCodec {
  public final TypeCodec<?> VAR_INT_CODEC = register(new VarIntCodec());
  public final TypeCodec<?> STRING_CODEC = register(new StringCodec());
  public final TypeCodec<?> UNSIGNED_SHORT_CODEC = register(new UnsignedShortCodec());
  public final TypeCodec<?> UUID_CODEC = register(new UUIDCodec());
  public final TypeCodec<?> LONG_CODEC = register(new LongCodec());

  public MinecraftDataCodec769() {
    super(Identifier.of("candlemc", "codec/769"));
  }
}

File: modules/protocol/769/src/main/kotlin/net/candlemc/protocol/codec/MinecraftDataCodec769.kt
package net.candlemc.protocol.codec

import net.candlemc.types.resource.Identifier

object MinecraftDataCodec769 : AbstractDataCodec(Identifier.of("candlemc", "codec/769")) {
    val VAR_INT_CODEC: TypeCodec<*> = register(VarIntCodec())!!
    val STRING_CODEC: TypeCodec<*> = register(StringCodec())!!
    val UNSIGNED_SHORT_CODEC: TypeCodec<*> = register(UnsignedShortCodec())!!
    val UUID_CODEC: TypeCodec<*> = register(UUIDCodec())!!
    val LONG_CODEC: TypeCodec<*> = register(LongCodec())!!
}
